### ISSUE TITLE
Add new `IndexOfString` method to DLL interfaces

### ISSUE THREAD

**Issue by delphidabbler** [delphidabbler/vilib#21]
While writing the demos, the absence of a method to get the index of a string within a string table became apparent.

`TVerInfoData` has such method, but it not surfaced by the DLL. This method takes a string table index and the string name as parameters.

If this method is added a new button to demonstrate it could be added to the demos. Demo docs would need to be updated too.

⭐ Note that any additions to the exposed interfaces would require v2 interfaces to be published in addition to the existing interfaces.

**Comment by delphidabbler** [comment]
If this method is added before the next release then this will require a minor version number bump.

**Comment by delphidabbler** [comment]
Decided to include in next release.

This is a backwards compatible API change, so requires a minor version number bump.

Don't forget the following:

> ⭐ Note that any additions to the exposed interfaces would require v2 interfaces to be published in addition to the existing interfaces.

And to update demo & demo docs.

**Comment by delphidabbler** [comment]
Define new interfaces as follows:

```pascal
type
  IVerInfoBinaryReader2 = interface(IVerInfoBinaryReader)
    ['{9f044ccd-2c9a-4fd9-a501-5ff868150e97}']
    function IndexOfString(TableIdx: Integer; const Name: WideString;
      out: Index: Integer): HResult; stdcall;

  IVerInfoBinary2 = interface(IVerInfoBinary)
    ['{7bdfb79a-70d9-4114-b00a-ef05b273ca93}']
    function IndexOfString(TableIdx: Integer; const Name: WideString;
      out: Index: Integer): HResult; stdcall;
  end;
```

We also need to alter the definition of `TVerInfoBinary` to:

```pascal
type
  TVerInfoBinary = class(TInterfacedObject,
    IUnknown,
    IVerInfoBinaryReader, IVerInfoBinaryReader2,
    IVerInfoBinary, IVerInfoBinary2)
  // ...
  end;
```

Finally, note that `CreateInstance` creates either `IVerInfoBinaryReader` or `IVerInfoBinary` instances that will need to be cast to `IVerInfoBinaryReader2` or `IVerInfoBinary2` as appropriate.

There are three possible solutions to this:

1. Do nothing. Require the user to cast to the v2 interfaces where required.
2. Provide v2 analogues to the existing CLSIDs that have the same effect but change `CreateInstance` to create v2 interfaced objects for the new CLSIDs.
3. Export a new `CreateInstance2` function that uses the existing CLSIDs but creates objects using the v2 interfaces.

**Comment by delphidabbler** [comment]
> Finally, note that `CreateInstance` creates either `IVerInfoBinaryReader` or `IVerInfoBinary` instances that will need to be cast to `IVerInfoBinaryReader2` or `IVerInfoBinary2` as appropriate.

Actually, `CreateInstance` happily takes a `IVerInfoBinaryReader2` or `IVerInfoBinary2` and returns what we'de expect, so no surgery required! :smiley:

**Comment by delphidabbler** [comment]
Implemented at commit 4fd8e7e
